pkg/hsperfdata/hsperfdata.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022-2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package hsperfdata
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+//HsperfdataPath returns the path of the hsperfdata file for a given pid,
+//it searches in all hsperfdata user directories, pids are assumed to be
+//unique regardless of username.
+//TODO(vthakkar): Add tests to makes sure above statement is true especially
+//with containers
+
+func HsperfdataPath(pid string) (string, error) {
+	perfGlob := filepath.Join(os.TempDir(), "hsperfdata_*", pid)
+	perfFiles, err := filepath.Glob(perfGlob)
+	if err != nil {
+		return "", err
+	}
+
+	if len(perfFiles) < 1 {
+		return "", fmt.Errorf("No hsperfdata file found for pid: %s", pid)
+	}
+
+	if len(perfFiles) > 1 {
+		return "", fmt.Errorf("More than one hsperfdata file found for pid: %s, this should never happen", pid)
+	}
+
+	filePath := perfFiles[0]
+
+	return filePath, nil
+}
+
+// HsperfdataPath returns a map(pid: hsperfdatafilepath)
+func HsperfdataPath(pids []string) (map[string]string, error) {
+	filePaths := make(map[string]string)
+	for _, pid := range pids {
+		filePath, err := HsperfdataPath(pid)
+		if err != nil {
+			return nil, err
+		}
+		filePaths[pid] = filePath
+	}
+
+	return filePaths, nil
+}
+
+// UserdataPaths returns all the java process hsperfdata paths belonging
+// to the given user
+func UserdataPaths(user string) (map[string]string, error) {
```

Comment:
```suggestion
func UserDataPaths(user string) (map[string]string, error) {
```
🔧 I think this has a different pattern than `hsperfdata`, so better to stick to conventions 😄